marimo/_plugins/ui/_impl/tables/types.py

```diff
@@ -11,3 +11,11 @@ class DataFrameLike(Protocol):
     def __dataframe__(
         self, nan_as_null: bool = False, allow_copy: bool = True
     ) -> DataFrame: ...
+
+
+def is_dataframe_like(value: object) -> bool:
+    return (
+        isinstance(value, DataFrameLike)
+        and hasattr(value, "__dataframe__")
+        and callable(value.__dataframe__)
```

Comment:
What if `__getattr__` itself returns callable?

Reply:
added more defensiveness: checking `value.__dataframe__ is value.__dataframe__` which could guard against some (but not all) of `__getattr__` returning a callable.

also added a check inside `DataFrameProtocolTableManager` after constructing the dataframe to check its shape further and falling back to `None`.